funcx_web_service/routes/funcx.py

```diff
@@ -80,9 +80,13 @@ def auth_and_launch(user_id, function_uuid, endpoints, input_data, app, token, s
        json object
     """
     # Check if the user is allowed to access the function
-    if not authorize_function(user_id, function_uuid, token):
+    try:
+        if not authorize_function(user_id, function_uuid, token):
+            return {'status': 'Failed',
+                    'reason': f'Unauthorized access to function: {function_uuid}'}
+    except Exception as e:
```

Comment:
It looks like `authorize_function` returns a bool based on whether the user owns the function, and if the function doesn't exist, raises an Exception. It would be cleaner to update the docstring for `authorize_function` to mention the exception, perhaps a `FunctionNotRegistered` exception, and report that specifically in the reason message.